influxdb3_telemetry/src/sender.rs

```diff
@@ -0,0 +1,55 @@
+use std::{sync::Arc, time::Duration};
+
+use observability_deps::tracing::{debug, error};
+
+use crate::store::{TelemetryPayload, TelemetryStore};
+use crate::{Result, TelemetryError};
+
+pub(crate) struct TelemetrySender {
+    client: reqwest::Client,
+    req_path: String,
+}
+
+impl TelemetrySender {
+    pub fn new(client: reqwest::Client, req_path: String) -> Self {
+        Self { client, req_path }
+    }
+
+    pub async fn try_sending(&self, telemetry: &TelemetryPayload) -> Result<()> {
+        debug!(telemetry = ?telemetry, "trying to send data to telemetry server");
+        let json = serde_json::to_vec(&telemetry).map_err(TelemetryError::CannotSerializeJson)?;
+        self.client
+            .post(self.req_path.as_str())
+            .body(json)
+            .send()
+            .await
+            .map_err(TelemetryError::CannotSendToTelemetryServer)?;
+        debug!("Successfully sent telemetry data to server");
+        Ok(())
+    }
+}
+
+pub(crate) async fn send_telemetry_in_background(
+    store: Arc<TelemetryStore>,
+    duration_secs: u64,
+) -> tokio::task::JoinHandle<()> {
+    tokio::spawn(async move {
+        let telem_sender = TelemetrySender::new(
+            reqwest::Client::new(),
+            "https://telemetry.influxdata.foo.com".to_owned(),
```

Comment:
This getting updated in a follow on PR?

Reply:
Yes - this will be updated in one of the follow on PRs